test/joiner_join_project_data_test.rb

```diff
@@ -36,5 +37,59 @@ def test_hide_hold_projects_in_public_mode
       Joiner.join_data @site
       assert_empty @site.data['projects']
     end
+
+    def project_data
+      { 'name' => 'test_fm',
+        'full_name' => 'project',
+        'github' => ['test_fm'],
+        'team' => %w(thing1 thing2 thing3),
+        'stack' => %w(thing4 thing5),
+        'owner_type' => 'project',
+      }
+    end
+
+    def project_data_with_errors
+      with_errors = project_data
+      project_data['errors'] = %w(error1 error2)
+      with_errors
+    end
+
+    def error_joiner_impl(errors)
```

Comment:
You can use `errors: {}`, I believe, if you want to avoid having to specify the empty hash explicitly. Up to you.